Sources/Utility/PkgConfig.swift

```diff
@@ -18,15 +18,69 @@ public enum PkgConfigError: Swift.Error {
     case nonWhitelistedFlags(String)
 }
 
-/// Get search paths from pkg-config itself.
-///
-/// This is needed because on Linux machines, the search paths can be different
-/// from the standard locations that we are currently searching.
-private let pkgConfigSearchPaths: [AbsolutePath] = {
-    let searchPaths = try? Process.checkNonZeroExit(
-        args: "pkg-config", "--variable", "pc_path", "pkg-config").chomp()
-    return searchPaths?.split(separator: ":").map({ AbsolutePath(String($0)) }) ?? []
-}()
+public struct PkgConfigExecutionDiagnostic: DiagnosticData {
+    public static let id = DiagnosticID(
+        type: AnyDiagnostic.self,
+        name: "org.swift.diags.pkg-config-execution",
+        defaultBehavior: .warning,
+        description: {
+            $0 <<< "Problem executing pkg-config"
+        }
+    )
+}
+
+struct PCFileFinder {
```

Comment:
This can be fileprivate, right?

Reply:
I think I need it to stay `internal` so the tests can access it, otherwise I'm getting:

```
jmsmith@SFO-M-JSMITH01 ~/w/B/a/swift-package-manager (yasumoto-master) [1]> swift test --parallel
Compile Swift Module 'Utility' (19 sources)
Compile Swift Module 'UtilityTests' (11 sources)
/Users/jmsmith/workspace/Baseline/apple/swift-package-manager/Tests/UtilityTests/PkgConfigParserTests.swift:72:46: error: use of unresolved identifier 'PCFileFinder'
        XCTAssertEqual("/custom/foo.pc", try PCFileFinder(diagnostics: diagnostics).locatePCFile(name: "foo", customSearchPaths: [AbsolutePath("/custom")], fileSystem: fs).asString)
                                             ^~~~~~~~~~~~
/Users/jmsmith/workspace/Baseline/apple/swift-package-manager/Tests/UtilityTests/PkgConfigParserTests.swift:74:57: error: use of unresolved identifier 'PCFileFinder'
        XCTAssertEqual("/usr/lib/pkgconfig/foo.pc", try PCFileFinder(diagnostics: diagnostics).locatePCFile(name: "foo", customSearchPaths: [], fileSystem: fs).asString)
                                                        ^~~~~~~~~~~~
error: terminated(1): /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2018-06-07-a.xctoolchain/usr/bin/swift-build-tool -f /Users/jmsmith/workspace/Baseline/apple/swift-package-manager/.build/debug.yaml test output:
```

(I read through [the Access Control](https://docs.swift.org/swift-book/LanguageGuide/AccessControl.html) docs and I thought `@testable` should've made this work, but that doesn't appear to be the case)

Reply:
Ah, I didn't realize there were unit test for this. `internal` makes sense then!